src/sentry/static/sentry/app/views/organizationIntegration/integrationRepos.jsx

```diff
@@ -0,0 +1,291 @@
+      success: data => {
+        let itemList = this.state.itemList;
+        itemList.forEach(item => {
+          if (item.id === data.id) {
+            item.status = data.status;
+          }
+        });
+        this.setState({itemList});
+      },
+      error: () => {
+        addErrorMessage(t('Unable to delete repository.'));
+      },
+      complete: () => {
+        IndicatorStore.remove(indicator);
+      },
+    });
+  };
+
+  cancelDelete = repo => {
+    let {orgId} = this.props;
+    let indicator = IndicatorStore.add(t('Saving changes..'));
+    this.api.request(`/organizations/${orgId}/repos/${repo.id}/`, {
+      method: 'PUT',
+      data: {status: 'visible'},
+      success: data => {
+        let itemList = this.state.itemList;
+        itemList.forEach(item => {
+          if (item.id === data.id) {
+            item.status = data.status;
+          }
+        });
+        this.setState({itemList});
+      },
+      error: () => {
+        addErrorMessage(t('An error occurred.'));
+      },
+      complete: () => {
+        IndicatorStore.remove(indicator);
+      },
+    });
+  };
+
+  renderDropdown() {
+    let access = new Set(this.context.organization.access);
+    if (!access.has('org:write')) {
+      return (
+        <DropdownButton
+          disabled={true}
+          title={t('You do not have permission to add repos')}
+          isOpen={false}
+          size="xsmall"
+        >
+          {t('Add Repo')}
+        </DropdownButton>
+      );
+    }
+    let repositories = this.state.integrationRepos.repos;
+    let items = (repositories || []).map(repo => {
+      return {
+        searchKey: `${repo.name}`,
+        value: `${repo.full_name}`,
+        label: (
+          <StyledListElement>
+            <StyledName>{repo.name}</StyledName>
+          </StyledListElement>
+        ),
+      };
+    });
+
+    let menuHeader = <StyledReposLabel>{t('Repositories')}</StyledReposLabel>;
+
+    return (
+      <DropdownAutoComplete
+        items={items}
+        onSelect={this.addRepo.bind(this)}
+        menuHeader={menuHeader}
+        emptyMessage={t('No repositories available')}
+      >
+        {({isOpen}) => (
+          <DropdownButton isOpen={isOpen} size="xsmall">
+            {t('Add Repo')}
+          </DropdownButton>
+        )}
+      </DropdownAutoComplete>
+    );
+  }
+
+  renderBody() {
+    const itemList = this.getIntegrationRepos() || [];
+    const header = (
+      <PanelHeader hasButtons>
+        <div>{t('Repositories')}</div>
+        <div style={{textTransform: 'none'}}>{this.renderDropdown()}</div>
+      </PanelHeader>
+    );
+
+    return (
+      <Panel>
+        {header}
+        <PanelBody>
+          {itemList.length === 0 && (
+            <Box>
+              <EmptyMessage size="large">{t('No Repositories Added')}</EmptyMessage>
+            </Box>
+          )}
+          {itemList.length > 0 && (
+            <Box>
+              {itemList.map(repo => {
```

Comment:
Only minor comment here is that you could consolidate the `Box` so that you don't have these extra `itemList` length checks.

```jsx
<Box>
  {itemList.length ===0 && (
     <EmptyMessage size="large">{t('No Repositories Added')}</EmptyMessage>
  )}
  {itemList.map( ... )}
</Box>
```

Since the map won't render anything if it's empty.

Reply:
I'll add that to https://github.com/getsentry/sentry/pull/8709